File: Programmers/Level2/프린터/프린터.py
from collections import deque
def solution(priorities, location):
    answer = 0
    deq = deque([[value,index] for index,value in enumerate(priorities)])
    while deq:
        tmp = deq.popleft()
        if deq and max(deq)[0] > tmp[0]:
            deq.append(tmp)
        else:
            answer += 1
            if location == tmp[1]:
                return answer

File: Programmers/Level2/프린터/프린터.java
import java.util.*;
class Solution {
   public int solution(int[] priorities, int location) {
        int answer = 1;

        PriorityQueue priority = new PriorityQueue<>(Collections.reverseOrder());

        for(int task : priorities){
            priority.add(task);
            //System.out.println(priority);
        }
        //{2,5,4,1,3};

        System.out.println(priority);
        while(!priority.isEmpty()){
            for(int i=0; i<priorities.length; i++){
                if(priorities[i] == (int)priority.peek()) {
                    if(i == location){
                        return answer;
                    }
                    priority.poll();
                    answer++;
                }
            }
        }

        return answer;
    }
}
